Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.Model.swift

```diff
@@ -0,0 +1,140 @@
+                        let geometry = decodable.geometry
+                        
+                        // We successfully decoded the vessel JSON so we should
+                        // add that vessel as a new observation.
+                        return .newObservation(
+                            geometry: Point(x: geometry.x, y: geometry.y, spatialReference: .wgs84),
+                            attributes: decodable.attributes.makeDictionary()
+                        )
+                    }
+                
+                return VesselFeed(events: feedEvents)
+            }
+            
+            let customDataSource = CustomDynamicEntityDataSource(info: info) { try makeFeed() }
+            
+            dynamicEntityLayer = DynamicEntityLayer(dataSource: customDataSource)
+            
+            let trackDisplayProperties = dynamicEntityLayer.trackDisplayProperties
+            trackDisplayProperties.showsPreviousObservations = true
+            trackDisplayProperties.showsTrackLine = true
+            trackDisplayProperties.maximumObservations = 20
+            
+            let labelDefinition = LabelDefinition(
+                labelExpression: SimpleLabelExpression(simpleExpression: "[VesselName]"),
+                textSymbol: TextSymbol(color: .red, size: 12)
+            )
+            labelDefinition.placement = .pointAboveCenter
+            
+            dynamicEntityLayer.addLabelDefinition(labelDefinition)
+            dynamicEntityLayer.labelsAreEnabled = true
+            
+            map.addOperationalLayer(dynamicEntityLayer)
+        }
+    }
+}
+
+private extension Array where Element == Field {
+    /// An array of fields that match the attributes of each observation in the data source.
+    ///
+    /// This schema is derived from the first row in the custom data source.
+    static var vesselFields: [Field] = [
+        Field(type: .text, name: "MMSI", alias: "MMSI", length: 256),
+        Field(type: .float64, name: "SOG", alias: "SOG", length: 8),
+        Field(type: .float64, name: "COG", alias: "COG", length: 8),
+        Field(type: .text, name: "VesselName", alias: "VesselName", length: 256),
+        Field(type: .text, name: "CallSign", alias: "CallSign", length: 256),
+        Field(type: .text, name: "globalid", alias: "globalid", length: 256)
+    ]
+}
+
+/// The vessel feed that is emitting custom dynamic entity events.
+private struct VesselFeed: CustomDynamicEntityFeed {
+    let events: AsyncPublisher<AnyPublisher<CustomDynamicEntityFeedEvent, Never>>
+    
+    /// Creates a feed that is receiving the events that are being passed in.
+    ///
+    /// The events will be emitted with a delay so it resembles live data.
+    /// - Parameter events: The events that will be emitted by the feed
+    init<S>(events: S) where S: Sequence, S.Element == CustomDynamicEntityFeedEvent {
+        self.events = events.publisher
+            .delay(for: 1, scheduler: RunLoop.main)
```

Comment:
I'll ask other platform to see how they emit the changes. Since there are a few thousands of entries, maybe we can make the delay smaller.

Reply:
William said they used 10ms delay.